engine/design_system.gemspec

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+lib = File.expand_path("lib", __dir__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require "design_system/version"
+
+Gem::Specification.new do |spec|
+  spec.name = "design_system"
```

Comment:
@pedantic-git I'm a little unsure about best practices for naming this.

At least for Rails engines the gem name and the engine namespace need to match. Using plain `design_system` here is nice as it means components can be called with `DesignSystem::PaginationComponent` which I like as an API.

I do wonder what happens if we published this more widely though. I assume best practice would involve prefixing this somehow. I have an aesthetic distaste for the `ca_` prefix we often use and that would end up with `CaDesignSystem::PaginationComponent` which…ugh. Alternatively `citizens_advice_design_system` would result in `CitizensAdviceDesignSystem::PaginationComponent` or even `CitizensAdvice::DesignSystem::PaginationComponent` which is accurate but…verbose.

Am I overthinking things? If we went for the prefixed name is there a way for apps to alias to a shorter name?

Reply:
Ended up looking around at other examples and went with `citizens_advice_components`, which makes the namespace `CitizensAdviceComponents::PreviewComponent`.

That extra `Component` at the end is the convention of view components. So a little verbose, but descriptive and includes the organisation name as a prefix. Feels OK.